### ISSUE TITLE
[NeuralNetworkReductor] training procedure

### ISSUE THREAD

**Issue by HenKlei** [HenKlei/pymor#9]
Multiple training runs should be performed and the one with the best result should be chosen as final neural network. The maximal number of trainings runs to perform is determined by a `restarts`-parameter. Further, the user should either decide on a basis size or on absolute/relative tolerances. In the latter case, the respective tolerance should be distributed evenly to the reduced basis error and the neural network error. Hence, the final neural network should produce an error on the validation set that lies below a prescribed threshold. Otherwise, no appropriate neural network was found and some kind of error should be thrown.

- [x] Multiple training runs; select the one with smallest validation error
- [x] Prescribe size of reduced basis
- [x] Prescribe error tolerances

**Comment by HenKlei** [comment]
In the case of error tolerances: should these tolerances be undercut on the validation set or the training set? At least the reduced basis error via POD is computed on the training set.

**Comment by HenKlei** [comment]
@sdrave There is one thing I'm currently not happy with: in the test case, the error of the basis is really small (1e-10 or so) if the basis size is 10. However, the neural network will not be able to reach such a small error. Nevertheless, the neural network will produce results with an average error of about 1e-5 on a test set. With the current implementation, such a neural network (which is not too bad I think) will be treated as being insufficient and no neural network will be returned at all by the reductor.

**Comment by sdrave** [comment]
Ok, sounds like we need more flexibility. Maybe add a parameter for the ann mean2 error which can be either set to a concrete value, `'like_basis'` in which case the POD mean2 error is used or to `None` which just takes the best net after the given number of restarts. I would like `'like_basis'` as the default. Or maybe it would better to let the default depend on what parameters are used to choose the basis size?

**Comment by HenKlei** [comment]
To me, the idea with an additional parameter sounds really good. This should be the most flexible way, I think. I'm going ahead and implement it the way you proposed.
